satpy/readers/fci_l1c_fdhsi.py

```diff
@@ -220,55 +268,75 @@ def _get_dataset_measurand(self, key, info=None):
 
         return res
 
-    def _get_dataset_quality(self, key, info=None):
-        """Load quality for channel.
-
-        Load a quality field for an FCI channel.  This is a bit involved in
-        case of FCI because each channel group (data/<channel>/measured) has
-        its own data variable 'pixel_quality', referred to in ancillary
-        variables (see also Satpy issue 1171), so some special treatment in
-        necessary.
-        """
-        # FIXME: replace by .removesuffix after we drop support for Python < 3.9
-        if key['name'].endswith("_pixel_quality"):
-            chan_lab = key['name'][:-len("_pixel_quality")]
-        else:
-            raise ValueError("Quality label must end with pixel_quality, got "
-                             f"{key['name']:s}")
-        grp_path = self.get_channel_measured_group_path(chan_lab)
+    def _get_dataset_quality(self, dsname):
+        """Load a quality field for an FCI channel."""
+        grp_path = self.get_channel_measured_group_path(_get_channel_name_from_dsname(dsname))
         dv_path = grp_path + "/pixel_quality"
         data = self[dv_path]
         return data
 
-    def get_channel_measured_group_path(self, channel):
+    def _get_dataset_index_map(self, dsname):
+        """Load the index map for an FCI channel."""
+        grp_path = self.get_channel_measured_group_path(_get_channel_name_from_dsname(dsname))
+        dv_path = grp_path + "/index_map"
+        data = self[dv_path]
+
+        data = data.where(data != data.attrs.get('_FillValue', 65535))
+        return data
+
+    @staticmethod
+    def _getitem(block, lut):
+        return lut[block.astype('uint16')]
+
+    def _get_dataset_aux_data(self, dsname):
+        """Get the auxiliary data arrays using the index map."""
+        # get index map
+        index_map = self._get_dataset_index_map(_get_channel_name_from_dsname(dsname))
+        # index map indexing starts from 1
+        index_map -= 1
+
+        # get lut values from 1-d vector
+        lut = self[AUX_DATA[_get_aux_data_name_from_dsname(dsname)]]
+
+        fv = default_fillvals.get(lut.dtype.str[1:], np.nan)
+        lut = lut.where(lut != fv)
+
```

Comment:
Could this be delegated to its own method?

Reply:
Done, thanks!